### PR TITLE
Add LM based dialogue acts extractor

### LINKED ISSUE #182: Add LM-based dialogue acts extractor
Add LM-based extractor capable of extraction multiple dialogue acts per utterance.

### PR DESCRIPTION
**Pull request by NoB0**
* Add a new dialogue acts extractor leveraging a language model
* Add prompts for dialogue acts extraction
* Add script to experiments with different LM and see their performances
* Add data and utils to perform experiments with LMs

Fixes #182

### CODE REVIEW

**Review by kbalog — APPROVED** [review]
LGTM with some comments for consideration

File: scripts/nlu/dialogue_acts_extraction.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    intent_error_rate,
+    slot_error_rate,
+)
+from usersimcrs.nlu.lm.lm_dialogue_act_extractor import LMDialogueActsExtractor
+
+
+def parse_gold_dialogue_acts(
+    json_dialogue_acts: List[Dict[str, Any]]
+) -> List[DialogueAct]:
+    """Parses gold dialogue acts from json format.
```
[the remark this answers is not in the document]
Comment (NoB0) [reply scripts/nlu/dialogue_acts_extraction.py:25]: Done.

File: scripts/nlu/metrics.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
+"""Define the evaluation metric for the dialogue act extraction.
+
+We define 5 metrics:
+1. Slot Error Rate (SER)
+2. Intent Error Rate (IER)
+3. Dialogue Acts Recall (DAR)
+4. Dialogue Acts Precision (DAP)
+5. Dialogue Acts F1 Score (DAF1)
```
[the remark this answers is not in the document]
Comment (NoB0) [reply scripts/nlu/metrics.py:8]: Done.

File: scripts/nlu/dialogue_acts_extraction.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        slot_values = json_dialogue_act.get("slot_values", [])
+        if slot_values:
+            for json_annotation in slot_values:
+                annotations.append(
+                    SlotValueAnnotation(
+                        json_annotation[0],
+                        json_annotation[1],
+                        json_annotation[2],
+                        json_annotation[3],
+                    )
```
[the remark this answers is not in the document]
Comment (NoB0) [reply scripts/nlu/dialogue_acts_extraction.py:46]: I noticed that this method is unused so I removed it.

File: scripts/nlu/metrics.py [hunk 4]
```diff
@@ -0,0 +1,143 @@
+"""Define the evaluation metric for the dialogue act extraction.
```
[the remark this answers is not in the document]
Comment (NoB0) [reply scripts/nlu/metrics.py:1]: Created a follow-up issue #185.

**Pull request merged**